### PR TITLE
Rename lib to src

### PR DESCRIPTION
**Pull request by greg0ire**
I am targetting this branch, because I assume people are using the provided Composer autoloader as opposed to `require_once`

## Changelog

```markdown
### Changed
- The directory structure is more standard
```

## Subject

`src` is more standard, so let's apply the principle of least
astonishment.

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Could you please add an upgrade note like done on exporter.

Another packages to move or is this the last one?

**Comment by greg0ire** [comment]
Nope : https://github.com/sonata-project/stats

**Comment by greg0ire** [comment]
Not sure we are supposed to touch this project though : last commit 3 years ago :confused:

**Comment by greg0ire** [comment]
@Soullivaneuh : I added the upgrade note

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "autoload": {
         "psr-4": {
             "Sonata\\Cache\\Tests\\": "test/",
```
Comment (core23) [inline composer.json]: Maybe add these to `autoload-dev`?

Comment (Soullivaneuh) [inline composer.json]: BC break risk. We are not sure no test class is used on other sonata projects.

**Comment by Soullivaneuh** [comment]
> Not sure we are supposed to touch this project though : last commit 3 years ago :confused:

cc @rande

File: composer.json (hunk 1 above)
Comment (greg0ire) [inline composer.json]: I'll make a major PR then.

**Comment by Soullivaneuh** [comment]
@greg0ire Please add this line on the upgrade:

> This should not change anything for users since this library is meant to be used with Composer autoloading only.

**Comment by greg0ire** [comment]
ping @sonata-project/contributors

**Comment by Soullivaneuh** [comment]
Thank you @greg0ire

**Comment by OskarStark** [comment]
Great job @greg0ire 🎉

**Pull request merged**